tokio/tests/rt_threaded.rs

```diff
@@ -322,6 +323,50 @@ fn multi_threadpool() {
     done_rx.recv().unwrap();
 }
 
+#[test]
+fn coop_and_block_in_place() {
```

Comment:
nit: it would be nice to have a comment summarizing what we're testing for here...

Reply:
It occurs to me that this is actually a racy test as well...

Reply:
fixed.